### PR TITLE
Delete ROX_VULN_REPORTING feature flag

### PR DESCRIPTION
**Pull request by pedrottimark**
## Description

Delete leftover feature flags is a prerequisite to replace `useFeatureFlagEnabled` with `useFeatureFlags` hook

1. pkg/features/list.go
    * Leave feature flag for now because some backend code still refers to `VulnReporting` /cc @clickboo

2. ui/apps/platform/src/Containers/MainPage/Body.tsx
    * Delete feature flag from conditional rendering for `VulnMgmtReports` element

3. ui/apps/platform/src/Containers/Navigation/NavigationSideBar/NavigationSideBar.tsx
    * Delete feature flag from conditional rendering for `vulnManagementReportsPath` path
    * Comment out `import` statement for `knownBackendFlags` as hint for people who add feature flags
    * Add disable comment for `isFeatureFlagEnabled` prop so `MainPage` continues to render the element with it

    Too bad, so sad prettier does not allow the following:

    ```tsx
    function NavigationSideBar({
        hasReadAccess,
        // isFeatureFlagEnabled,
    }: NavigationSideBarProps): ReactElement {
    ```

4. ui/apps/platform/src/Containers/SystemConfig/ConfigDataRetentionDetailWidget.tsx
    * Delete feature flags from conditional rendering for `expiredVulnReqRetentionDurationDays` property

5. ui/apps/platform/src/Containers/SystemConfig/SystemConfigForm/SystemConfigForm.tsx
    * Delete feature flags from conditional rendering for `expiredVulnReqRetentionDurationDays` property

6. ui/apps/platform/src/utils/featureFlags.ts
    * Delete frontend definition of feature flag

## Checklist
- [x] Investigated and inspected CI test results
- [x] ~Unit test and regression tests added~

### manual testing

/main/systemconfig
systemconfig

### integration testing

1. `yarn build` in ui
2. `yarn cypress` in ui/apps/platform
    * vulnmanagement/reporting/displayReports.test.js (does not depend on feature flag)
    * vulnmanagement/reporting/manageReports.test.js (does not depend on feature flag)

### CODE REVIEW

File: ui/apps/platform/src/Containers/Navigation/NavigationSideBar/NavigationSideBar.tsx [hunk 1]
```diff
@@ -4,7 +4,7 @@ import { Nav, NavList, NavExpandable, PageSidebar } from '@patternfly/react-core
 
 import { IsFeatureFlagEnabled } from 'hooks/useFeatureFlags';
 import { HasReadAccess } from 'hooks/usePermissions';
-import { knownBackendFlags } from 'utils/featureFlags';
+// import { knownBackendFlags } from 'utils/featureFlags';
```
Comment (vjwilson) [inline ui/apps/platform/src/Containers/Navigation/NavigationSideBar/NavigationSideBar.tsx:7]: If we are leaving this in for future use, let's add a comment above for why it is needed sometimes.

**Review by vjwilson — CHANGES_REQUESTED** [review]
UI part looks good.
(Request for one clarifying comment)

Also, need to remove from feature flag list in Go code.
```
	// VulnReporting enables scheduled vulnerability reporting workflow, that allows the creation and management of vulnerability reporting configurations.
	VulnReporting = registerFeature("Enable creation of scheduled vulnerability reports to be sent via email notifier", "ROX_VULN_REPORTING", true)
```
in `pkg/features/list.go`

**Comment by pedrottimark** [comment]
Occurrences of `VulnReporting` in backend code:

central/main.go
  344,14: 	if features.VulnReporting.Enabled() {
  715,14: 	if features.VulnReporting.Enabled() {

central/reports/manager/manager_impl.go
  68,15: 	if !features.VulnReporting.Enabled() {

central/reports/manager/singleton.go
  38,15: 	if !features.VulnReporting.Enabled() {

central/reports/scheduler/schedule.go
  377,15: 	if !features.VulnReporting.Enabled() {

central/role/datastore/singleton.go
  51,15: 		if features.VulnReporting.Enabled() {

central/role/resources/list.go
  73,130: 	VulnerabilityReports             = newResourceMetadataWithFeatureFlag("VulnerabilityReports", permissions.GlobalScope, features.VulnReporting)

pkg/features/list.go
  33,5: 	// VulnReporting enables scheduled vulnerability reporting workflow, that allows the creation and management of vulnerability reporting configurations.
  34,2: 	VulnReporting = registerFeature("Enable creation of scheduled vulnerability reports to be sent via email notifier", "ROX_VULN_REPORTING", true)

**Comment by vjwilson** [comment]
> Occurrences of `VulnReporting` in backend code:
> 
> central/main.go 344,14: if features.VulnReporting.Enabled() { 715,14: if features.VulnReporting.Enabled() {
> 
> central/reports/manager/manager_impl.go 68,15: if !features.VulnReporting.Enabled() {
> 
> central/reports/manager/singleton.go 38,15: if !features.VulnReporting.Enabled() {
> 
> central/reports/scheduler/schedule.go 377,15: if !features.VulnReporting.Enabled() {
> 
> central/role/datastore/singleton.go 51,15: if features.VulnReporting.Enabled() {
> 
> central/role/resources/list.go 73,130: VulnerabilityReports = newResourceMetadataWithFeatureFlag("VulnerabilityReports", permissions.GlobalScope, features.VulnReporting)
> 
> pkg/features/list.go 33,5: // VulnReporting enables scheduled vulnerability reporting workflow, that allows the creation and management of vulnerability reporting configurations. 34,2: VulnReporting = registerFeature("Enable creation of scheduled vulnerability reports to be sent via email notifier", "ROX_VULN_REPORTING", true)

@clickboo Can Mark remove these references to VM Reporting feature flag in this PR, so that we make the removal atomic?

**Comment by clickboo** [comment]
> > Occurrences of `VulnReporting` in backend code:
> > central/main.go 344,14: if features.VulnReporting.Enabled() { 715,14: if features.VulnReporting.Enabled() {
> > central/reports/manager/manager_impl.go 68,15: if !features.VulnReporting.Enabled() {
> > central/reports/manager/singleton.go 38,15: if !features.VulnReporting.Enabled() {
> > central/reports/scheduler/schedule.go 377,15: if !features.VulnReporting.Enabled() {
> > central/role/datastore/singleton.go 51,15: if features.VulnReporting.Enabled() {
> > central/role/resources/list.go 73,130: VulnerabilityReports = newResourceMetadataWithFeatureFlag("VulnerabilityReports", permissions.GlobalScope, features.VulnReporting)
> > pkg/features/list.go 33,5: // VulnReporting enables scheduled vulnerability reporting workflow, that allows the creation and management of vulnerability reporting configurations. 34,2: VulnReporting = registerFeature("Enable creation of scheduled vulnerability reports to be sent via email notifier", "ROX_VULN_REPORTING", true)
> 
> @clickboo Can Mark remove these references to VM Reporting feature flag in this PR, so that we make the removal atomic?

Absolutely! Let's do it. Do you want me to push a change this PR or you got it @pedrottimark ?

**Review by vjwilson — APPROVED** [review]

**Review by clickboo — APPROVED** [review]

**Pull request merged**